data/src/main/java/org/oppia/data/backends/gae/NetworkModule.kt

```diff
@@ -0,0 +1,63 @@
+package org.oppia.data.backends.gae
+
+import dagger.Module
+import dagger.Provides
+import okhttp3.OkHttpClient
+import org.oppia.data.backends.gae.api.ClassroomService
+import org.oppia.data.backends.gae.api.TopicService
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+/**
+ * Module which provides all required dependencies about network
+ *
+ * Sample resource: https://github.com/gahfy/Feed-Me/tree/unitTests
+ */
+@Module
+class NetworkModule {
+
+  @Qualifier
+  private annotation class OppiaRetrofit
+
+  /**
+   * Provides the Retrofit object.
```

Comment:
Optional nit: we probably don't need Javadocs for module methods since nobody is calling them directly. It's useful in cases with specifically complicated dependencies (e.g. if the injection site has to have some other configuration set up for the injection to succeed).

Reply:
For now I am keeping this Javadoc.